test/cache-interceptor/cache-tests.mjs

```diff
@@ -0,0 +1,346 @@
+  }
+  if (environment.opts.type) {
+    console.log(`           type: ${styleText('gray', environment.opts.type)}`)
+  }
+  if (environment.ignoredTests) {
+    console.log(`  ignored tests: ${styleText('gray', JSON.stringify(environment.ignoredTests))}`)
+  }
+
+  try {
+    await runTests(environment)
+  } catch (err) {
+    console.error(err)
+  }
+
+  const stats = printResults(environment, getResults())
+  printStats(stats)
+
+  // Cleanup state
+  for (const key of Object.keys(testUUIDs)) {
+    delete testUUIDs[key]
+  }
+  for (const key of Object.keys(testResults)) {
+    delete testResults[key]
+  }
+
+  console.log('')
+}
+
+/**
+ * @param {number} idx
+ * @param  {TestEnvironment[][]} testOptions
+ * @returns {TestEnvironment[]}
+ */
+function buildTestEnvironments (idx, testOptions) {
+  const baseEnvironments = testOptions[idx]
+
+  if (idx + 1 >= testOptions.length) {
+    // We're at the end, nothing more to make a matrix out of
+    return baseEnvironments
+  }
+
+  /**
+   * @type {TestEnvironment[]}
+   */
+  const environments = []
+
+  // Get all of the environments below us
+  const subEnvironments = buildTestEnvironments(idx + 1, testOptions)
+
+  for (const baseEnvironment of baseEnvironments) {
+    const combinedEnvironments = subEnvironments.map(subEnvironment => {
+      const ignoredTests = baseEnvironment.ignoredTests ?? []
+      if (subEnvironment.ignoredTests) {
+        ignoredTests.push(...subEnvironment.ignoredTests)
+      }
+
+      return {
+        opts: {
+          ...baseEnvironment.opts,
+          ...subEnvironment.opts
+        },
+        ignoredTests: ignoredTests.length > 0 ? ignoredTests : undefined
+      }
+    })
+
+    environments.push(...combinedEnvironments)
+  }
+
+  return environments
+}
+
+/**
+ * @param {TestEnvironment[]} environments
+ * @returns {TestEnvironment[]}
+ */
+function filterEnvironments (environments) {
+  const { values } = CLI_OPTIONS
+
+  if (values.type) {
+    environments = environments.filter(env =>
+      env.opts.type === undefined ||
+      values.type?.includes(env.opts.type)
+    )
+  }
+
+  return environments
+}
+
+/**
+ * @param {TestEnvironment} environment
+ */
+async function runTests (environment) {
+  // Start the test server. We use a worker here since the suite doesn't expose it
+  const worker = new Worker(join(import.meta.dirname, 'cache-tests-worker.mjs'), {
+    env: {
+      npm_config_protocol: PROTOCOL,
+      npm_config_port: `${PORT}`,
+      npm_config_pidfile: PIDFILE
+    }
+  })
+
+  try {
+    await Promise.race([
+      once(worker, 'message'),
+      new Promise((_resolve, reject) => {
+        setTimeout(() => {
+          reject(new Error('worker thread startup timed out'))
+        }, 5000)
+      })
+    ])
+
+    const client = new Agent().compose(interceptors.cache(environment.opts))
+    setGlobalDispatcher(client)
+
+    // Run the tests
+    await runTestSuite(tests, fetch, true, BASE_URL)
```

Comment:
the test suite is still using the global fetch, not the one that undici exports.